Test failure in FORECAST

Folllowing up from #1624.

On current master: 

```
dipy (master)$nosetests dipy/reconst/tests/test_forecast.py 
F.....
======================================================================
FAIL: dipy.reconst.tests.test_forecast.test_forecast_positive_constrain
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/arokem/.virtualenvs/dipy/lib/python3.7/site-packages/nose/case.py", line 198, in runTest
    self.test(*self.arg)
  File "/Users/arokem/.virtualenvs/dipy/lib/python3.7/site-packages/numpy/testing/_private/decorators.py", line 155, in skipper_func
    return f(*args, **kwargs)
  File "/Users/arokem/source/dipy/dipy/reconst/tests/test_forecast.py", line 53, in test_forecast_positive_constrain
    assert_almost_equal(fodf[fodf < 0].sum(), 0, 2)
  File "/Users/arokem/.virtualenvs/dipy/lib/python3.7/site-packages/numpy/testing/_private/utils.py", line 584, in assert_almost_equal
    raise AssertionError(_build_err_msg())
AssertionError: 
Arrays are not almost equal to 2 decimals
 ACTUAL: -0.015931951384411995
 DESIRED: 0

----------------------------------------------------------------------
Ran 6 tests in 0.264s

```

Looking closer, it looks like indeed the positivity constraint is no longer applied to the fodf. That is, there are negative components in the fodf. 

My guess is that this has to do with the (very) recent version of cvxpy that was just released today. My hunch is that this is a bug in the way the positivity constraints are implemented in this new version of cvxpy, rather than a change in API. It would be good to come up with a minimal example that we could report to them.

--- a/dipy/reconst/forecast.py
+++ b/dipy/reconst/forecast.py
@@ -480,8 +480,8 @@ def lb_forecast(sh_order):
     diag_lb = np.zeros(n_c)
     counter = 0
     for l in range(0, sh_order + 1, 2):
-        for _ in range(-l, l + 1):
-            diag_lb[counter] = (l * (l + 1)) ** 2
-            counter += 1
+        stop = 2 * l + 1 + counter
+        diag_lb[counter:stop] = (l * (l + 1)) ** 2
+        counter = stop
 
     return np.diag(diag_lb)

